nil pointer panic on recent version

### What platforms are affected?

macos

### What architectures are affected?

arm64

### What SpiceDB version are you using?

v1.45.0

### Steps to Reproduce

1. Start docker image with `latest`
2. Run with args `["spicedb", "serve-testing", "--grpc-addr", ":50053", "--readonly-grpc-addr", ":50054", "--otel-endpoint", "''", "--load-configs", "/schema/authzed.yml"]`
3. I'm currently running spicedb inside docker. Would have to investigate if this is required for it to fail.
### Expected Result

spicedb runs.

### Actual Result

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3fa328]

goroutine 1 [running]:
net/http.(*Client).do(0x0, 0x4000bfa000)
        /opt/hostedtoolcache/go/1.24.4/x64/src/net/http/client.go:603 +0x178
net/http.(*Client).Do(...)
        /opt/hostedtoolcache/go/1.24.4/x64/src/net/http/client.go:587
github.com/authzed/spicedb/pkg/releases.getLatestReleaseWithClient({0x31a5728, 0x4000463260}, 0x0)
        /home/runner/work/spicedb/spicedb/pkg/releases/releases.go:48 +0x158
github.com/authzed/spicedb/pkg/releases.GetLatestRelease({0x31a5728?, 0x4000463260?})
        /home/runner/work/spicedb/spicedb/pkg/releases/releases.go:35 +0x28
github.com/authzed/spicedb/pkg/releases.CheckIsLatestVersion({0x31a5728, 0x4000463260}, 0x4000463270?, 0x2e1d090)
        /home/runner/work/spicedb/spicedb/pkg/releases/versions.go:57 +0x88
github.com/authzed/spicedb/pkg/releases.mustCheckAndLogVersionUpdate({0x31a5728, 0x4000463260}, 0x77359400?, 0x31f9540?)
        /home/runner/work/spicedb/spicedb/pkg/releases/cli.go:25 +0x24
github.com/authzed/spicedb/pkg/cmd/server.DefaultPreRunE.CheckAndLogRunE.func9(0x4000b86908?, {0x207400?, 0x40007e5bc8?, 0x0?})
        /home/runner/work/spicedb/spicedb/pkg/releases/cli.go:80 +0x80
github.com/authzed/spicedb/pkg/cmd/server.DefaultPreRunE.CommandStack.func11(0x4000b86908, {0x4000b13a80, 0x0, 0x8})
        /home/runner/go/pkg/mod/github.com/jzelinskie/cobrautil/v2@v2.0.0-20240819150235-f7fe73942d0f/cobrautil.go:78 +0x78
github.com/spf13/cobra.(*Command).execute(0x4000b86908, {0x4000b13a00, 0x8, 0x8})
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1000 +0x7ec
github.com/spf13/cobra.(*Command).ExecuteC(0x4000329508)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x384
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071
main.main()
        /home/runner/work/spicedb/spicedb/cmd/spicedb/main.go:125 +0x188
```

--- a/pkg/releases/releases.go
+++ b/pkg/releases/releases.go
@@ -36,6 +36,9 @@ func GetLatestRelease(ctx context.Context) (*Release, error) {
 }
 
 func getLatestReleaseWithClient(ctx context.Context, httpClient *http.Client) (*Release, error) {
+	if httpClient == nil {
+		httpClient = http.DefaultClient
+	}
 	url := fmt.Sprintf("https://api.github.com/repos/%s/%s/releases/latest", githubNamespace, githubRepository)
 
 	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)

